### ISSUE TITLE
Suggestion: documentation about PCI passthrough and upgrades

### ISSUE THREAD

**Issue by hkparker** [xcp-ng/xcp#414]
Hey team,

I recently did an upgrade from 7.6 to 8.1, and had some issues with PCI passthrough that were entirely my fault (#412).  It occurred to me however that it would be helpful to have some documentation in the upgrade informing the user that they will need to re-add the kernel boot parameters for the pciback devices.  Maybe this is actually something that can be done as part of the upgrade process automatically, but in my case the device addresses changed so it's probably safer to leave it up to the user to ensure their PCI passthrough is still how they want it after an upgrade.

Let me know what you think.

**Comment by rjt** [comment]
kexec reboot allows one to restart the kernel without resetting the hardware.   I have kexec reboot on servers that take forever to initialize hardware and finally boot up.   But have never done it with XEN.  Is it possible that kexec reboot could be used after translating a device name to a PCI address? 

Actually, if during the upgrade process and  if passthru is enabled for any VM, then regenerate from lspci | egrep?

Just thinking out loud.

**Comment by stormi** [comment]
Hi. So the upgrade does not retain the `xen-pciback.hide` parameter?

**Comment by nagilum99** [comment]
I'm irritated by: "xen_pciback: Error parsing pci_devs_to_hide at "(0000:82.0.0)" (from linked #412. Was supposed to be 0000:82:00.0". If the first one worked before, it would mean that something has changed inside the kernel or device handling.

**Comment by hkparker** [comment]
@stormi IIRC it didn't, no.  I should have taken better notes, I was doing a lot, but I'm pretty confident it didn't.  I ended up needing to change the addresses anyway, though I'm not clear if that's because I swapped a card in the same slot, or if the addresses were mapped differently by the new kernel, so maybe it makes sense to not retain them and just instruct the user to follow up on this after an update.

@nagilum99 oh it never worked before like that.  That was a typo on my part when I first attempted to add the hide parameter back in.
